src/DataWrangling/EN4/EN4.jl

```diff
@@ -0,0 +1,257 @@
+module EN4
+
+export EN4Metadatum, EN4_field, EN4_mask, EN4_immersed_grid, adjusted_EN4_tracers, initialize!
+export EN4Monthly
+export EN4FieldTimeSeries, EN4Restoring, LinearlyTaperedPolarMask
+
+using ClimaOcean
+using ClimaOcean.DataWrangling
+using ClimaOcean.DataWrangling: inpaint_mask!, NearestNeighborInpainting, download_progress, compute_native_date_range
+using ClimaOcean.InitialConditions: three_dimensional_regrid!, interpolate!
+
+using Oceananigans
+using Oceananigans: location
+using Oceananigans.Architectures: architecture, child_architecture
+using Oceananigans.BoundaryConditions
+using Oceananigans.DistributedComputations
+using Oceananigans.DistributedComputations: DistributedField, all_reduce, barrier!
+using Oceananigans.Utils
+
+using KernelAbstractions: @kernel, @index
+using NCDatasets
+using JLD2
+using Downloads: download
+using Dates
+using Adapt
+using Scratch
+
+download_EN4_cache::String = ""
+function __init__()
+    global download_EN4_cache = @get_scratch!("EN4")
+end
+
+include("EN4_metadata.jl")
+include("EN4_mask.jl")
+
+# Vertical coordinate
+const EN4_z = [
+    -5500.0,
+    -5200.5986,
+    -4901.459,
+    -4602.695,
+    -4304.469,
+    -4007.0154,
+    -3710.6636,
+    -3415.8828,
+    -3123.3313,
+    -2833.926,
+    -2548.9243,
+    -2270.0168,
+    -1999.4135,
+    -1739.8875,
+    -1494.7288,
+    -1267.542,
+    -1061.846,
+    -880.5102,
+    -725.1709,
+    -595.8501,
+    -490.9494,
+    -407.6244,
+    -342.3651,
+    -291.5638,
+    -251.9186,
+    -220.6418,
+    -195.5097,
+    -174.8189,
+    -157.3019,
+    -142.0345,
+    -128.3522,
+    -115.7825,
+    -103.9913,
+    -92.7437,
+    -81.8753,
+    -71.2711,
+    -60.8509,
+    -50.5586,
+    -40.3553,
+    -30.214,
+    -20.1158,
+    -10.0475,
+    -0.0,
+]
+
+empty_EN4_field(variable_name::Symbol; kw...) = empty_EN4_field(Metadatum(variable_name, dataset=EN4Monthly()); kw...)
+
+function empty_EN4_field(metadata::EN4Metadata;
+                          architecture = CPU(), 
+                          horizontal_halo = (7, 7))
+
+    Nx, Ny, Nz, _ = size(metadata)
+    loc = location(metadata)
+    longitude = (0, 360)
+    latitude = (-90, 90)
+    TX, TY = (Periodic, Bounded)
+
+    if variable_is_three_dimensional(metadata)
+        TZ = Bounded
+        LZ = Center
+        z = EN4_z
```

Comment:
is this the line that needs to be different from `empty_ECCO_field`?

Reply:
could we define a function called `z_faces(metadata)`, and then change the name of the parent method to `empty_field`? Then we only need one method `empty_field`, which finds the z faces for that metadata

Reply:
I can do that. Shall I @taimoorsohail?

Reply:
Yes, ideally for EN4 we would want something like:

```Julia
function z_faces(m::EN4Metadata)
netcdf_filename = metadata_filename(metadata)[1] # index 1 because it will be a string of filenames for all dates
data = Dataset(netcdf_filename) 
z_faces = sort(-1.*Array(data["depth_bnds"])[1,:]) # index 1 because this is the lower bound
push!(z_faces, -1.*Array(data["depth_bnds"])[2,end]) #append the last term of index 2, should be -5500 m in EN4
return z_faces
```

Reply:
I tried using the above to replace 

https://github.com/CliMA/ClimaOcean.jl/blob/b0c141c9eddeb38bc4163112675cee2251209c6f/src/DataWrangling/EN4/EN4.jl#L39

but it wasn't working. I think it must be something trivial, but I'll stop for today.